nomad/resource_variable.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVariableWrite,
+		Update: resourceVariableWrite,
+		Delete: resourceVariableDelete,
+		Read:   resourceVariableRead,
+		Exists: resourceVariableExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"namespace": {
+				Description: "Variable namespace filter",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     api.DefaultNamespace,
+			},
+			"items": {
+				Type:      schema.TypeMap,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceVariableWrite(d *schema.ResourceData, meta any) error {
+	client := meta.(ProviderConfig).client
+
+	variable := api.Variable{
+		Namespace: d.Get("namespace").(string),
+		Path:      d.Get("path").(string),
+		Items:     make(map[string]string),
+	}
+	for name, value := range d.Get("items").(map[string]any) {
+		variable.Items[name] = value.(string)
```

Comment:
Hey @lgfa29,

The namespace resource does the same by the looks of it:
```go
	for name, value := range d.Get("meta").(map[string]interface{}) {
		m[name] = value.(string)
	}
```

So it seems to me that if this was a concern it would be something that should be solved holistically for the provider.